main/src/cgeo/geocaching/filters/core/DateRangeGeocacheFilter.java

```diff
@@ -34,6 +37,15 @@ public void setMinMaxDate(final Date min, final Date max) {
         this.dateFilter.setMinMaxDate(min, max);
     }
 
+    public void setRelativeMinMaxDays(final int daysBeforeToday, final int daysAfterToday) {
```

Comment:
The filter does not store whether the dates were set relative or absolute. This will lead to a strange situation: suppose today is the 12.6.22 and you set relative ranges to "previous 10 days until today". Then the filter would store the absolute min date 2.6.22 and the absolute may date 12.6.22. if you store this filter (e.g. as a named filter or current live filter or current offline filter) and open it e.g. 2 weeks later at 26.6.22, it would STILL filter for range 2.-12.6.22, although the user would expect it to filter for 16.-26.6. at this point in time.

I am afraid a flag and maybe the relative filter span needs to be stored and retrieved also from filter config to achieve the desired effect. Or am I missing something?

Reply:
Hm, true... But as I see it to get the existing functionality of the DateRange filter combined with a relative range would require a helper filter that constructs a temporary DateRange filter (that's not added to the stored filter) whenever it's instantiated.

Reply:
My idea was: 
* the class DateFilter currently has two attributes: minDate and maxDate. This is enough to define absolute ranges
* One could add three more attributes: isRelative (boolean), minDaysToSubstract (int), maxDaysToAdd (int)
*  If flag "isRelative" is false then everything stays as is. If it is true, then all usages of "minDate" are replaces with "DateUtils.addDays(new Date(), -minDaysToSubstract), same for "maxDate"
* The configuration stored (done in "setConfig"/"getConfig") stores/retrieves the three new attributes as well

But I admit, this would be a little work to do...